presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -453,9 +453,11 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
 
             boolean containsEligibleTableBucket = false;
```

Comment:
nit: I think we can eliminate `containsEligibleTableBucket` now (and use `!eligibleTableBucketNumbers.isEmpty()`)

Reply:
Maybe also add a `checkState(tableBucketCount >= readBucketCount)`. This works as comments to help reasoning the code :)

Reply:
Both comments make sense. Will fix the code